PostHog/Replay/PostHogReplayIntegration.swift

```diff
@@ -225,29 +255,63 @@
 
         private func isAssetsImage(_ image: UIImage) -> Bool {
             // https://github.com/daydreamboy/lldb_scripts#9-pimage
+            // do not mask if its an asset image, likely not PII anyway
             image.imageAsset?.value(forKey: "_containingBundle") != nil
         }
 
         private func isAnyInputSensitive(_ view: UIView) -> Bool {
-            config.sessionReplayConfig.maskAllTextInputs || config.sessionReplayConfig.maskAllImages || view.isNoCapture()
+            isTextInputSensitive(view) || config.sessionReplayConfig.maskAllImages
         }
 
         private func isTextInputSensitive(_ view: UIView) -> Bool {
             config.sessionReplayConfig.maskAllTextInputs || view.isNoCapture()
         }
 
+        private func isLabelSensitive(_ view: UILabel) -> Bool {
+            isTextInputSensitive(view) && hasText(view.text)
+        }
+
+        private func isButtonSensitive(_ view: UIButton) -> Bool {
+            isTextInputSensitive(view) && hasText(view.titleLabel?.text)
+        }
+
         private func isTextViewSensitive(_ view: UITextView) -> Bool {
-            isTextInputSensitive(view) || view.isSensitiveText()
+            (isTextInputSensitive(view) || view.isSensitiveText()) && hasText(view.text)
+        }
+
+        private func isPickerSensitive(_ view: UIPickerView) -> Bool {
+            isTextInputSensitive(view) && view.dataSource != nil
+        }
+
+        private func isSwitchSensitive(_ view: UISwitch) -> Bool {
+            var containsText = true
+            if #available(iOS 14.0, *) {
+                containsText = hasText(view.title)
+            }
+
+            return isTextInputSensitive(view) && containsText
         }
 
         private func isTextFieldSensitive(_ view: UITextField) -> Bool {
-            isTextInputSensitive(view) || view.isSensitiveText()
+            (isTextInputSensitive(view) || view.isSensitiveText()) && (hasText(view.text) || hasText(view.placeholder))
+        }
+
+        private func hasText(_ text: String?) -> Bool {
+            if let text = text, !text.isEmpty {
+                return true
+            } else {
+                // if there's no text, there's nothing to mask
+                return false
+            }
         }
 
         private func isImageViewSensitive(_ view: UIImageView) -> Bool {
             var isAsset = false
             if let image = view.image {
                 isAsset = isAssetsImage(image)
+            } else {
+                // if there's no image, there's nothing to mask
+                return false
```

Comment:
Even if masking is enabled, only mask if there's actually any content (text|image)